internal/querycoord/task_scheduler.go

```diff
@@ -638,8 +640,35 @@ func (scheduler *TaskScheduler) scheduleLoop() {
 			if triggerTask.getState() == taskUndo || triggerTask.getState() == taskDoing {
 				err = scheduler.processTask(triggerTask)
 				if err != nil {
-					log.Warn("scheduleLoop: process triggerTask failed", zap.Int64("triggerTaskID", triggerTask.getTaskID()), zap.Error(err))
-					alreadyNotify = false
+					// Checks in preExecute not passed,
+					// for only LoadCollection & LoadPartitions
+					if errors.Is(err, ErrCollectionLoaded) ||
+						errors.Is(err, ErrLoadParametersMismatch) {
+						log.Warn("scheduleLoop: preExecute failed",
+							zap.Int64("triggerTaskID", triggerTask.getTaskID()),
+							zap.Error(err))
+
+						triggerTask.setState(taskExpired)
+						if errors.Is(err, ErrLoadParametersMismatch) {
+							triggerTask.setState(taskFailed)
+						}
+
+						triggerTask.notify(err)
+
+						err = removeTaskFromKVFn(triggerTask)
+						if err != nil {
```

Comment:
NIT: feels weird that processTask error need to treat some kind of error differently. Any way to merge all kind of errors here?